presto-native-execution/presto_cpp/main/runtime-metrics/prometheus-reporter/tests/PrometheusReporterTest.cpp

```diff
@@ -0,0 +1,154 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "presto_cpp/main/runtime-metrics/prometheus-reporter/PrometheusStatsReporter.h"
+
+#include <gtest/gtest.h>
+
+namespace facebook::presto::prometheus {
+class PrometheusReporterTest : public testing::Test {
+ public:
+  void SetUp() override {
+    reporter = std::make_shared<PrometheusStatsReporter>(testLabels);
+  }
+  void verifySerializedResult(
+      const std::string& fullSerializedResult,
+      std::vector<std::string>& expected) {
+    auto i = 0;
+    std::stringstream ss(fullSerializedResult);
+    std::string line;
+    while (getline(ss, line, '\n')) {
+      EXPECT_EQ(line, expected[i++]);
+    }
+  }
+  const std::map<std::string, std::string> testLabels = {
+      {"cluster", "test_cluster"},
+      {"worker", "test_worker_pod"}};
+  const std::string labelsSerialized =
+      R"(cluster="test_cluster",worker="test_worker_pod")";
+  std::shared_ptr<PrometheusStatsReporter> reporter;
+};
+
+TEST_F(PrometheusReporterTest, testCountAndGauge) {
+  reporter->registerMetricExportType(
+      "test.key1", facebook::velox::StatType::COUNT);
+  reporter->registerMetricExportType(
+      "test.key2", facebook::velox::StatType::AVG);
+  reporter->registerMetricExportType(
+      "test.key3", facebook::velox::StatType::SUM);
+  reporter->registerMetricExportType(
+      "test.key4", facebook::velox::StatType::RATE);
+  EXPECT_EQ(
+      facebook::velox::StatType::COUNT,
+      reporter->registeredMetricsMap_.find("test.key1")->second.statType);
+  EXPECT_EQ(
+      facebook::velox::StatType::AVG,
+      reporter->registeredMetricsMap_.find("test.key2")->second.statType);
+  EXPECT_EQ(
+      facebook::velox::StatType::SUM,
+      reporter->registeredMetricsMap_.find("test.key3")->second.statType);
+  EXPECT_EQ(
+      facebook::velox::StatType::RATE,
+      reporter->registeredMetricsMap_.find("test.key4")->second.statType);
+
+  std::vector<size_t> testData = {10, 11, 15};
+  for (auto i : testData) {
+    reporter->addMetricValue("test.key1", i);
+    reporter->addMetricValue("test.key2", i + 1000);
+  }
+  // Uses default value of 1 for second parameter.
+  reporter->addMetricValue("test.key1");
+  auto fullSerializedResult = reporter->getSerializedMetrics();
```

Comment:
This function doesn't exist anymore. Should be replaced with `fetchMetrics`?
Also on line 110.